modules/planning/on_lane_planning.cc

```diff
@@ -504,14 +503,12 @@ Status OnLanePlanning::Plan(
 }
 
 bool OnLanePlanning::CheckPlanningConfig(const PlanningConfig& config) {
-  if (!config.has_standard_planning_config()) {
-    return false;
-  }
-  if (config.standard_planning_config()
-          .planner_public_road_config()
-          .scenario_type_size() == 0) {
+  if (!config.has_standard_planning_config() ||
+      config.standard_planning_config().planner_public_road_config()
+                                       .scenario_type_size() == 0) {
```

Comment:
1. No need to check `has_XXX` if we'll check its sub-field is empty. The default value is empty.
2. `scenario_type_size() == 0` -> `scenario_type().empty()`